examples/drivers/windows/otLwf/tunnel.c

```diff
@@ -0,0 +1,1745 @@
+    IPV6_HEADER* v6Header;
+
+    UNREFERENCED_PARAMETER(Secure); // TODO - What should we do with unsecured packets?
+
+    NetBufferList =
+        NdisAllocateNetBufferAndNetBufferList(
+            pFilter->NetBufferListPool,     // PoolHandle
+            0,                              // ContextSize
+            0,                              // ContextBackFill
+            NULL,                           // MdlChain
+            0,                              // DataOffset
+            0                               // DataLength
+            );
+    if (NetBufferList == NULL)
+    {
+        status = STATUS_INSUFFICIENT_RESOURCES;
+        LogWarning(DRIVER_DEFAULT, "Failed to create command NetBufferList");
+        goto exit;
+    }
+
+    // Set the flag to indicate its a IPv6 packet
+    NdisSetNblFlag(NetBufferList, NDIS_NBL_FLAGS_IS_IPV6);
+    NET_BUFFER_LIST_INFO(NetBufferList, NetBufferListFrameType) =
+        UlongToPtr(RtlUshortByteSwap(ETHERNET_TYPE_IPV6));
+        
+    // Initialize NetBuffer fields
+    NetBuffer = NET_BUFFER_LIST_FIRST_NB(NetBufferList);
+    NET_BUFFER_CURRENT_MDL(NetBuffer) = NULL;
+    NET_BUFFER_CURRENT_MDL_OFFSET(NetBuffer) = 0;
+    NET_BUFFER_DATA_LENGTH(NetBuffer) = 0;
+    NET_BUFFER_DATA_OFFSET(NetBuffer) = 0;
+    NET_BUFFER_FIRST_MDL(NetBuffer) = NULL;
+    
+    // Allocate the NetBuffer for NetBufferList
+    if (NdisRetreatNetBufferDataStart(NetBuffer, BufferLength, 0, NULL) != NDIS_STATUS_SUCCESS)
+    {
+        NetBuffer = NULL;
+        status = STATUS_INSUFFICIENT_RESOURCES;
+        LogError(DRIVER_DEFAULT, "Failed to allocate NB for command NetBufferList, %u bytes", BufferLength);
+        goto exit;
+    }
+
+    // Get the pointer to the data buffer for the header data
+    DataBuffer = (PUCHAR)NdisGetDataBuffer(NetBuffer, BufferLength, NULL, 1, 0);
+    NT_ASSERT(DataBuffer);
+    
+    // Copy the data over
+    RtlCopyMemory(DataBuffer, Buffer, BufferLength);
+
+    v6Header = (IPV6_HEADER*)DataBuffer;
+    
+    // Filter messages to addresses we expose
+    if (!IN6_IS_ADDR_MULTICAST(&v6Header->DestinationAddress) &&
+        otLwfFindCachedAddrIndex(pFilter, &v6Header->DestinationAddress) == -1)
+    {
+        LogVerbose(DRIVER_DATA_PATH, "Filter: %p dropping internal address message.", pFilter);
+        goto exit;
+    }
+    
+    // Filter internal Thread messages
+    /*if (v6Header->NextHeader == IPPROTO_UDP &&
+        BufferLength >= sizeof(IPV6_HEADER) + sizeof(UDPHeader) &&
+        memcmp(&pFilter->otLinkLocalAddr, &v6Header->DestinationAddress, sizeof(IN6_ADDR)) == 0)
+    {
+        // Check for MLE message
+        UDPHeader* UdpHeader = (UDPHeader*)(v6Header + 1);
+        if (UdpHeader->DestinationPort == UdpHeader->SourcePort &&
+            UdpHeader->DestinationPort == RtlUshortByteSwap(19788)) // MLE Port
+        {
+            LogVerbose(DRIVER_DATA_PATH, "Filter: %p dropping MLE message.", pFilter);
+            goto exit;
+        }
+    }*/
```

Comment:
We are trying to avoid commented-out code blocks unless accompanied with a well-articulated reason for why it should be kept around and under what circumstances it should be removed or uncommented.